pkgs/tools/typesetting/tectonic/wrapper.nix

```diff
@@ -0,0 +1,61 @@
+{ lib
+, makeBinaryWrapper
+, symlinkJoin
+, tectonic
+, biberVersions
+}:
+
+let
+
+  pname = "tectonic-with-biber";
+  biber = biberVersions.biber-for-tectonic;
+  inherit (biber) version;
+
+  ## manually construct `name` for `symlinkJoin`
+  name = "${pname}-${version}";
+
+  meta = tectonic.meta // {
+    inherit name;
+    description = "Modernized TeX/LaTeX engine, with biber for bibliography";
+    longDescription = ''
+      This package wraps tectonic with biber without triggering rebuilds.
+      The biber executable is exposed with a version suffix, such as
+      `biber-${version}`, to prevent conflict with the `biber` bundled with
+      texlive in nixpkgs. Example use:
+
+          let
+            pkgs = <...>; # import nixpkgs here
+          in pkgs.tectonic-with-biber.override {
+            biber = pkgs.biberVersions.biber-for-tectonic;
+          }
+
+      This serves as a fix for:
+      - https://github.com/tectonic-typesetting/tectonic/issues/893
+    '';
+    maintainers = with lib.maintainers; [ bryango ];
+  };
+
+  ## produce the correct `meta.position` for `symlinkJoin`
+  pos = builtins.unsafeGetAttrPos "description" meta;
+
+in
+
+symlinkJoin {
+
+  inherit pname version name meta pos;
+
+  paths = [ tectonic ];
+  nativeBuildInputs = [ makeBinaryWrapper ];
+  passthru = { biber = biber; };
+
+  ## tectonic runs biber when it detects it needs to run it, see:
+  ## https://github.com/tectonic-typesetting/tectonic/releases/tag/tectonic%400.7.0
+  postBuild = ''
+    wrapProgram $out/bin/tectonic \
+      --prefix PATH : "${lib.getBin biber}/bin"
+    makeBinaryWrapper "${lib.getBin biber}/bin/biber" \
+      $out/bin/biber-${biber.version}
+  '';
+  ## the biber executable is exposed as `biber-${biber.version}`
```

Comment:
Oh, I think it would be better to expose biber in path; in my own experience, I would often run biber alone (without tectonic) to debug the bib entries. But if this is confusing, then I am okay to drop it. Please advise.